### PR TITLE
[SHELL32] Fix for incorrect init of hide/show hidden files & folders (CORE-16932)

### PR DESCRIPTION
**Pull request by KyleKatarn31**
## Purpose

_ In current implementation, init value for hide/show hidden files & folders is set to 1 (Show) while "default" button sets it to 2 (Hide). "Default" button should set to "Show" (1).

JIRA issue: [CORE-16932](https://jira.reactos.org/browse/CORE-16932)

## Proposed changes

_ Change of Advanced Files & Folders Settings action of Default button to select "Show"
_ Change is made by flipping the HKLM default values

image

### CODE REVIEW

**Comment by Extravert-ir** [comment]
I'd like the default to remain as it is

**Comment by KyleKatarn31** [comment]
> 
> 
> I'd like the default to remain as it is

Reason for change :
Current init :
- Is not the safest option for users and is not aligned with Windows practices
- Is not aligned with the "Default" button on the advanced settings page (Files & Folders options)

**Comment by learn-more** [comment]
We are nowhere near a place that we can update features to be convenient for users.
At the moment it makes sense that they are convenient for developers.

**Comment by KyleKatarn31** [comment]
> 
> 
> We are nowhere near a place that we can update features to be convenient for users.
> At the moment it makes sense that they are convenient for developers.

Yes. That's the purpose of the "new" proposed design.

**Comment by SergeGautherie** [comment]
Add a comment somewhere, to remember to change it back in the future?

**Comment by KyleKatarn31** [comment]
> 
> 
> Add a comment somewhere, to remember to change it back in the future?

I thought that you all wanted this to be "definitive". Do you want me to refer PR+CORE-xxxx along with the impacted lines in comments?

**Comment by KyleKatarn31** [comment]
> 
> 
> Add a comment somewhere, to remember to change it back in the future?

Comments added. Would you please Review & Approve if OK ?

File: boot/bootdata/hivedef.inf [hunk 1]
```diff
@@ -1895,8 +1895,9 @@ HKCU,"SOFTWARE\Microsoft\Windows\CurrentVersion\Explorer\CLSID\{20D04FE0-3AEA-10
 HKCU,"SOFTWARE\Microsoft\Windows\CurrentVersion\Explorer\Advanced","ListviewShadow",0x00010003,0x00000001
 HKCU,"SOFTWARE\Microsoft\Windows\CurrentVersion\Explorer\Advanced","HideFileExt",0x00010003,0x00000000
 HKCU,"SOFTWARE\Microsoft\Windows\CurrentVersion\Explorer\Advanced","StartMenuLogoff",0x00010003,0x00000001
-HKCU,"SOFTWARE\Microsoft\Windows\CurrentVersion\Explorer\Advanced","Hidden",0x00010003,1
+HKCU,"SOFTWARE\Microsoft\Windows\CurrentVersion\Explorer\Advanced","Hidden",0x00010003,1
```
Comment (SergeGautherie) [inline boot/bootdata/hivedef.inf:1898]: Why add an ending space?

File: boot/bootdata/hivesft.inf [hunk 2]
```diff
@@ -317,19 +317,21 @@ HKLM,"SOFTWARE\Microsoft\Windows\CurrentVersion\Explorer\Advanced\Folder\Hidden\
 HKLM,"SOFTWARE\Microsoft\Windows\CurrentVersion\Explorer\Advanced\Folder\Hidden\NOHIDDEN","Type",0x00000000,"radio"
 HKLM,"SOFTWARE\Microsoft\Windows\CurrentVersion\Explorer\Advanced\Folder\Hidden\NOHIDDEN","CheckedValue",0x00010001,0x00000002
 HKLM,"SOFTWARE\Microsoft\Windows\CurrentVersion\Explorer\Advanced\Folder\Hidden\NOHIDDEN","ValueName",0x00000000,"Hidden"
-HKLM,"SOFTWARE\Microsoft\Windows\CurrentVersion\Explorer\Advanced\Folder\Hidden\NOHIDDEN","DefaultValue",0x00010001,0x00000002
+HKLM,"SOFTWARE\Microsoft\Windows\CurrentVersion\Explorer\Advanced\Folder\Hidden\NOHIDDEN","DefaultValue",0x00010001,0x00000001
 HKLM,"SOFTWARE\Microsoft\Windows\CurrentVersion\Explorer\Advanced\Folder\Hidden\NOHIDDEN","HKeyRoot",0x00010001,0x80000001
 HKLM,"SOFTWARE\Microsoft\Windows\CurrentVersion\Explorer\Advanced\Folder\Hidden\NOHIDDEN","HelpID",0x00000000,"shell.hlp#51104"
+; "DefaultValue" to be changed to 2 if we later want to have "Hide hidden files by default"
```
Comment (SergeGautherie) [inline boot/bootdata/hivesft.inf:323]: Comments are usually inserted just before the line they apply to, don't they?

File: boot/bootdata/hivedef.inf (hunk 1 above)
Comment (KyleKatarn31) [reply boot/bootdata/hivedef.inf:1898]: OK, i fix

File: boot/bootdata/hivesft.inf (hunk 2 above)
Comment (KyleKatarn31) [reply boot/bootdata/hivesft.inf:323]: OK, i fix

**Comment by Extravert-ir** [comment]
Unless we are not going to add documentation for all registry variables here, I don't see a point in having comments only for some of them.
Default settings is actually distributive-dependent. Windows Server and Workstation have different settings, for example

**Comment by KyleKatarn31** [comment]
> 
> 
> Unless we are not going to add documentation for all registry variables here, I don't see a point in having comments only for some of them.
> Default settings is actually distributive-dependent. Windows Server and Workstation have different settings, for example

I understood the request from other dev as a must since we do something one way while ReactOS is mainly for dev while on the long run we want to be able to come back to the other way with minimum efforts.

**Comment by learn-more** [comment]
For the one merging this: Please note that the commit message needs to be fixed...

**Pull request merged**